src/config.ts

```diff
@@ -216,13 +216,16 @@ function getProvider(
   config: Record<string, any>,
   metamask: ethers.providers.JsonRpcProvider | null,
 ): ethers.providers.JsonRpcProvider {
-  // Use Alchemy in production. Otherwise use Metamask if installed.
   if (import.meta.env.PROD) {
     return new ethers.providers.AlchemyWebSocketProvider(network.name, config.alchemy.key);
   } else if (metamask) {
     return metamask;
+  } else if (import.meta.env.DEV) {
+    // We are using here ethers.js Infura project id, which is used by ethers.providers.getDefaultProvider()
+    // https://github.com/ethers-io/ethers.js/blob/4a8d579dcaf026d0c232e20176605d34cba4767d/packages/providers/src.ts/infura-provider.ts#L17
+    return new ethers.providers.InfuraProvider(network.name, "84842078b09946638c03157f83405213");
```

Comment:
I wouldn't do this because it'll break when the key is revoked, which we have no control over..

Can we not call `ethers.getDefaultProvider()` or something?

Reply:
Yeah that's true, the issue is that `ethers.getDefaultProvider()` returns a `BaseProvider` while we require a `JsonRpcProvider` will look into how we could still make it work.

Reply:
Ok in the end we should be able to use `getDefaultProvider()` and cast the result to `JsonRpcProvider`

The only caveat there is that we won't have a `send` method available, which we use in the `utils.getTokens` function.
Nevertheless this function is almost exclusively for Alchemy since neither Metamask nor WalletConnect can query `alchemy_*` endpoints.